### PR TITLE
Replace rdp_constant: np.ndarray -> Dict[str, np.ndarray] 

### PR DESCRIPTION
**Pull request by IonesioJunior**
## Description
Fixes: #7138

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
Existing Publish tests

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Review by IshanMi — CHANGES_REQUESTED** [review]
Looks good to me! Before we merge could we add a unit test to make sure the `first_try_branch` method is updating the ledgers as expected?

File: packages/syft/src/syft/core/adp/data_subject_ledger.py [hunk 1]
```diff
@@ -115,43 +115,40 @@ def get_unique_data_subjects(data_subjects_query: np.ndarray) -> np.ndarray:
...
-    max_entity = len(unique_data_subjects)
 
-    input_entities_indexes_list: List[np.ndarray] = []
-
-    for data_subject_idx, data_subject in enumerate(unique_data_subjects):
+    for data_subject_name in unique_data_subjects:
         # Create a mask where the current data subject is present
-        data_subject = DataSubjectArray([data_subject])
+        data_subject = DataSubjectArray([data_subject_name])
         ds_mask = np.isin(data_subject_array, data_subject)
```
Comment (teo-milea) [inline packages/syft/src/syft/core/adp/data_subject_ledger.py:128]: I dont remember if @shubham3121 merged his PR but I recall there was a problem with this np.isin for nested data_subject in groups

**Review by teo-milea — CHANGES_REQUESTED** [review]
just a small change, apart from that LGTM! Of course, this will have to be tested again with the new publishing just to make sure everything works before the release

File: packages/syft/src/syft/core/adp/data_subject_ledger.py (hunk 1 above)
Comment (shubham3121) [reply packages/syft/src/syft/core/adp/data_subject_ledger.py:128]: @teo-milea that change is merged in your fix publish loop branch. So it should be fine once that PR is merged.

Comment (teo-milea) [reply packages/syft/src/syft/core/adp/data_subject_ledger.py:128]: great! sorry that I forgot :laughing:

**Pull request merged**